libraries/SuplaDevice/src/supla/network/network.cpp

```diff
@@ -103,15 +108,39 @@ void message_received(void *_srpc,
         ((SuplaDeviceClass *)_sdc)->onGetUserLocaltimeResult(rd.data.sdc_user_localtime_result);
         break;                                                 
       }                                                         
+      case SUPLA_SD_CALL_DEVICE_CALCFG_REQUEST: {
+        TDS_DeviceCalCfgResult result;
+        result.ReceiverID = rd.data.sd_device_calcfg_request->SenderID;
+        result.ChannelNumber = rd.data.sd_device_calcfg_request->ChannelNumber;
+        result.Command = rd.data.sd_device_calcfg_request->Command;
+        result.Result = SUPLA_CALCFG_RESULT_NOT_SUPPORTED;
+        result.DataSize = 0;
+
+        if (rd.data.sd_device_calcfg_request->SuperUserAuthorized != 1) {
```

Comment:
You have assumed in advance that all CALCFG calls will need authorization. It doesn't always have to be this way.

Reply:
Is there any way to automatically detect if authorization is needed? Or does it have to be "switch-case" based on request->Command ?

Or can we server simplify it and fill "1" here for all messages that doesn't require authorization?